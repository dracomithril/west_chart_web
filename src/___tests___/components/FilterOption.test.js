/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import FilteringOption from "./../../components/FilteringOptions/FilterOption";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from "enzyme-to-json";
import configureMockStore from "redux-mock-store";

Enzyme.configure({ adapter: new Adapter() });
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<FilteringOption/>', () => {
  let elem = {
    control: { name: "date_create_control", id: 'create' },
    input: { max: 31, name: "create_control" },
    valueName: "created_time",
    description: { start: 'created in last ', end: ' days' },
    type: 'countDays'
  };
  it('renders without crashing ChartPresenter', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <FilteringOption {...elem} />, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
  it('should be able to change state of woc checkbox', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <FilteringOption {...elem} />, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    const wocCheckbox = wrapper.findWhere(n => n.props().id === 'create').first();
    expect(wocCheckbox.props().checked).toBeTruthy();
    wocCheckbox.simulate('change', { target: { id: 'create', checked: false } });
    let actions = store.getActions();
    expect(actions[0].type).toBe('TOGGLE_FILTER');
    expect(actions[0].id).toBe('create');
    expect(actions[0].checked).toBeFalsy();

  })

});
