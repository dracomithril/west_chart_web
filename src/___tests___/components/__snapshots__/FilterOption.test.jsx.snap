// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilteringOption/> renders without crashing FilteringOption 1`] = `
<div>
  <WithStyles(Checkbox)
    color="primary"
    id="create_checkbox"
    name="date_create_control"
    onChange={[Function]}
  />
  created in last 
  <input
    className="filter-option__days"
    id="create_input"
    min={0}
    onChange={[Function]}
    step={1}
    type="number"
  />
   days
</div>
`;

exports[`<FilteringOption/> should be able to change state of woc checkbox 1`] = `
<div>
  <WithStyles(Checkbox)
    checked={true}
    color="primary"
    id="create_checkbox"
    name="date_create_control"
    onChange={[Function]}
  />
  created in last 
  <input
    className="filter-option__days"
    id="create_input"
    min={0}
    onChange={[Function]}
    step={1}
    type="number"
  />
   days
</div>
`;
